File: public/images/stageicons/download_icons.py
import os
import requests
from bs4 import BeautifulSoup
from urllib.parse import urljoin

def download_svgs():
    # Create directory if it doesn't exist
    os.makedirs(os.path.dirname(os.path.abspath(__file__)), exist_ok=True)
    
    # URL containing the SVG files
    base_url = 'https://tecrider.com/images/items/'
    
    # Get the page content
    response = requests.get(base_url)
    soup = BeautifulSoup(response.text, 'html.parser')
    
    # Find all links that end with .svg
    svg_links = [link.get('href') for link in soup.find_all('a') if link.get('href', '').endswith('.svg')]
    
    print(f"Found {len(svg_links)} SVG files to download...")
    
    # Download each SVG file
    for link in svg_links:
        filename = os.path.basename(link)
        full_url = urljoin(base_url, link)
        
        print(f"Downloading {filename}...")
        
        # Get the SVG content
        svg_response = requests.get(full_url)
        
        # Save the file
        with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), filename), 'wb') as f:
            f.write(svg_response.content)
            
    print("Download complete!")

if __name__ == "__main__":
    try:
        download_svgs()
    except Exception as e:
        print(f"An error occurred: {str(e)}") 

File: public/images/stageicons/download_icons.js
const fs = require('fs');
const path = require('path');
const fetch = require('node-fetch');
const cheerio = require('cheerio');

async function downloadSvgs() {
    // Create directory if it doesn't exist
    const dir = path.dirname(__filename);
    if (!fs.existsSync(dir)) {
        fs.mkdirSync(dir, { recursive: true });
    }

    // URL containing the SVG files
    const baseUrl = 'https://tecrider.com/images/items/';

    try {
        // Get the page content
        const response = await fetch(baseUrl);
        const html = await response.text();
        const $ = cheerio.load(html);

        // Find all links that end with .svg
        const svgLinks = $('a')
            .map((_, el) => $(el).attr('href'))
            .get()
            .filter(href => href && href.endsWith('.svg'));

        console.log(`Found ${svgLinks.length} SVG files to download...`);

        // Download each SVG file
        for (const link of svgLinks) {
            const filename = path.basename(link);
            const fullUrl = new URL(link, baseUrl).toString();

            console.log(`Downloading ${filename}...`);

            // Get the SVG content
            const svgResponse = await fetch(fullUrl);
            const svgContent = await svgResponse.buffer();

            // Save the file
            fs.writeFileSync(path.join(dir, filename), svgContent);
        }

        console.log('Download complete!');
    } catch (error) {
        console.error('An error occurred:', error.message);
    }
}

downloadSvgs(); 